src/Spice86.Core/Emulator/CPU/CfgCpu/Ast/Parser/AstExpressionParser.cs

```diff
@@ -0,0 +1,316 @@
+            }
+        }
+        return left;
+    }
+
+    private ValueNode ParseShift() {
+        ValueNode left = ParseAdditive();
+        while (true) {
+            if (Match("<<")) {
+                left = new BinaryOperationNode(DataType.UINT32, left, BinaryOperation.LEFT_SHIFT, ParseAdditive());
+            } else if (Match(">>")) {
+                left = new BinaryOperationNode(DataType.UINT32, left, BinaryOperation.RIGHT_SHIFT, ParseAdditive());
+            } else {
+                break;
+            }
+        }
+        return left;
+    }
+
+    private ValueNode ParseAdditive() {
+        ValueNode left = ParseMultiplicative();
+        while (true) {
+            if (CurrentChar() == '+') {
+                Advance();
+                left = new BinaryOperationNode(DataType.UINT32, left, BinaryOperation.PLUS, ParseMultiplicative());
+            } else if (CurrentChar() == '-' && !char.IsDigit(PeekChar())) {
+                Advance();
+                left = new BinaryOperationNode(DataType.UINT32, left, BinaryOperation.MINUS, ParseMultiplicative());
+            } else {
+                break;
+            }
+        }
+        return left;
+    }
+
+    private ValueNode ParseMultiplicative() {
+        ValueNode left = ParseUnary();
+        while (true) {
+            if (CurrentChar() == '*') {
+                Advance();
+                left = new BinaryOperationNode(DataType.UINT32, left, BinaryOperation.MULTIPLY, ParseUnary());
+            } else if (CurrentChar() == '/') {
+                Advance();
+                left = new BinaryOperationNode(DataType.UINT32, left, BinaryOperation.DIVIDE, ParseUnary());
+            } else if (CurrentChar() == '%') {
+                Advance();
+                left = new BinaryOperationNode(DataType.UINT32, left, BinaryOperation.MODULO, ParseUnary());
+            } else {
+                break;
+            }
+        }
+        return left;
+    }
+
+    private ValueNode ParseUnary() {
+        if (CurrentChar() == '!') {
+            Advance();
+            return new UnaryOperationNode(DataType.BOOL, UnaryOperation.NOT, ParseUnary());
+        }
+        if (CurrentChar() == '-') {
+            Advance();
+            return new UnaryOperationNode(DataType.UINT32, UnaryOperation.NEGATE, ParseUnary());
+        }
+        if (CurrentChar() == '~') {
+            Advance();
+            return new UnaryOperationNode(DataType.UINT32, UnaryOperation.BITWISE_NOT, ParseUnary());
+        }
+        return ParsePrimary();
+    }
+
+    private ValueNode ParsePrimary() {
+        SkipWhitespace();
+
+        // Parentheses
+        if (CurrentChar() == '(') {
+            Advance();
+            ValueNode expr = ParseExpression();
+            if (CurrentChar() != ')') {
+                throw new ArgumentException("Expected ')'");
+            }
+            Advance();
+            return expr;
+        }
+
+        // Memory access: byte[...], word[...], dword[...]
```

Comment:
Add support for segmented pointers.
Change the syntax to match what is produced by the AstInstructionRenderer.